***
# <font color=green size=10>CURSO DE ESTATÍSTICA - PARTE 2</font>
***

## Trabalho sobre Probabilidades, Amostragem e Estimações

Utilizando os conhecimentos adquiridos em nosso treinamento execute as tarefas abaixo. Siga o roteiro proposto e vá completando as células vazias.

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# <font color=green>DATASET DO PROJETO</font>
***

### Pesquisa Nacional por Amostra de Domicílios - 2015

A <b>Pesquisa Nacional por Amostra de Domicílios - PNAD</b> investiga anualmente, de forma permanente, características gerais da população, de educação, trabalho, rendimento e habitação e outras, com periodicidade variável, de acordo com as necessidades de informação para o país, como as características sobre migração, fecundidade, nupcialidade, saúde, segurança alimentar, entre outros temas. O levantamento dessas estatísticas constitui, ao longo dos 49 anos de realização da pesquisa, um importante instrumento para formulação, validação e avaliação de políticas orientadas para o desenvolvimento socioeconômico e a melhoria das condições de vida no Brasil.

### Fonte dos Dados

https://ww2.ibge.gov.br/home/estatistica/populacao/trabalhoerendimento/pnad2015/microdados.shtm

### Variáveis utilizadas

> ### Renda
> ***

Rendimento mensal do trabalho principal para pessoas de 10 anos ou mais de idade.

> ### Idade
> ***

Idade do morador na data de referência em anos.

> ### Altura (elaboração própria)
> ***

Altura do morador em metros.

> ### UF
> ***

|Código|Descrição|
|---|---|
|11|Rondônia|
|12|Acre|
|13|Amazonas|
|14|Roraima|
|15|Pará|
|16|Amapá|
|17|Tocantins|
|21|Maranhão|
|22|Piauí|
|23|Ceará|
|24|Rio Grande do Norte|
|25|Paraíba|
|26|Pernambuco|
|27|Alagoas|
|28|Sergipe|
|29|Bahia|
|31|Minas Gerais|
|32|Espírito Santo|
|33|Rio de Janeiro|
|35|São Paulo|
|41|Paraná|
|42|Santa Catarina|
|43|Rio Grande do Sul|
|50|Mato Grosso do Sul|
|51|Mato Grosso|
|52|Goiás|
|53|Distrito Federal|

> ### Sexo	
> ***

|Código|Descrição|
|---|---|
|0|Masculino|
|1|Feminino|

> ### Anos de Estudo
> ***

|Código|Descrição|
|---|---|
|1|Sem instrução e menos de 1 ano|
|2|1 ano|
|3|2 anos|
|4|3 anos|
|5|4 anos|
|6|5 anos|
|7|6 anos|
|8|7 anos|
|9|8 anos|
|10|9 anos|
|11|10 anos|
|12|11 anos|
|13|12 anos|
|14|13 anos|
|15|14 anos|
|16|15 anos ou mais|
|17|Não determinados| 
||Não aplicável|

> ### Cor
> ***

|Código|Descrição|
|---|---|
|0|Indígena|
|2|Branca|
|4|Preta|
|6|Amarela|
|8|Parda|
|9|Sem declaração|

#### <font color='red'>Observação</font>
***
> Os seguintes tratamentos foram realizados nos dados originais:
> 1. Foram eliminados os registros onde a <b>Renda</b> era inválida (999 999 999 999);
> 2. Foram eliminados os registros onde a <b>Renda</b> era missing;
> 3. Foram considerados somente os registros das <b>Pessoas de Referência</b> de cada domicílio (responsável pelo domicílio).

***
***

### Utilize a célula abaixo para importar as bibliotecas que precisar para executar as tarefas
#### <font color='red'>Sugestões: pandas, numpy, scipy etc.</font>

In [ ]:
import pandas as pd
import numpy as np


In [ ]:
from scipy.stats import norm
from scipy.stats import poisson
from scipy.stats import binom

### Importe o dataset e armazene o conteúdo em uma DataFrame

In [ ]:
dados = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Alura - DataScience Primeiros passos/Estatística com Python parte 2: probabilidade e amostragem/Final do curso/dados.csv')
dados.head()

UF  Sexo  Idade  Cor  Anos de Estudo  Renda    Altura
0  11     0     23    8              12    800  1.603808
1  11     1     23    2              12   1150  1.739790
2  11     1     35    8              15    880  1.760444
3  11     0     46    2               6   3500  1.783158
4  11     1     47    8               9    150  1.690631

### Visualize o conteúdo do DataFrame

In [ ]:
dados.head()

UF  Sexo  Idade  Cor  Anos de Estudo  Renda    Altura
0  11     0     23    8              12    800  1.603808
1  11     1     23    2              12   1150  1.739790
2  11     1     35    8              15    880  1.760444
3  11     0     46    2               6   3500  1.783158
4  11     1     47    8               9    150  1.690631

# <font color='green'>Problema A</font>

Avaliando nosso dataset é possível verificar que a **proporção de homens** como chefes de domicílios é de quase **70%**. Precisamos **selecionar aleatoriamente grupos de 10 indivíduos** para verificar as diferenças entre os rendimentos em cada grupo. Qual a **probabilidade de selecionamos um grupo que apresente a mesma proporção da população**, ou seja, selecionarmos um grupo que seja **composto por 7 homens e 3 mulheres**?

#### <font color='blue'>Como tarefa extra, verifique a real proporção de homens e mulheres em nosso dataset (vimos como fazer isso em nosso primeiro curso de estatística).</font>

#### <font color='red'>Verifique que tipo de distribuição de probabilidade se encaixa neste experimento.</font>

### Solução

In [ ]:
proporcao_por_sexo = pd.DataFrame(dados.Sexo.value_counts())
proporcao_por_sexo['Proporcao'] = dados.Sexo.value_counts(normalize = True) * 100
proporcao_por_sexo.rename(index = {0: 'Homem', 1: 'Mulher'}, columns = {'Sexo':'Frequencia'}, inplace = True)
proporcao_por_sexo

Frequencia  Proporcao
Homem        53250  69.299844
Mulher       23590  30.700156

Acima temos um dataset composto pela frequencia e a proporção real de homens e mulheres no nosso dataset principal


In [ ]:
p = 0.7 #Probabilidade de ser Homem
p

0.7

In [ ]:
n = 10 # Tamanho da amostra
n

10

In [ ]:
k = 7 # Qntd de homens igual a da proporção
k

7

In [ ]:
probabilidade = binom.pmf(k=k, n=n, p = p)
probabilidade

0.26682793200000016

A probabilidade de seleionarmos um grupo com a mesma proporção do dataset é de aproximadamente 26,68%.

# <font color='green'>Problema B</font>

Ainda sobre a questão anterior, **quantos grupos de 10 indivíduos** nós precisaríamos selecionar, de forma aleatória, para conseguir **100 grupos compostos por 7 homens e 3 mulheres**?

#### <font color='red'>Lembre-se da forma de cálculo da média de uma distribuição binomial</font>

### Solução

In [ ]:
n_grupos = 100 / probabilidade
int(n_grupos.round())

375

Serão necessários selecionar aproximadamente 375 grupos para conseguir 100 grupos compostos por 7 homens e 3 mulheres 

# <font color='green'>Problema C</font>

Um cliente nos encomendou um estudo para avaliar o **rendimento dos chefes de domicílio no Brasil**. Para isso precisamos realizar uma nova coleta de dados, isto é, uma nova pesquisa de campo. Após reunião com o cliente foi possível elencar o seguinte conjunto de informações:

> A. O resultado da pesquisa precisa estar pronto em **2 meses**;

> B. Teremos somente **R$\$$ 150.000,00** de recursos para realização da pesquisa de campo; e
    
> C. Seria interessante uma **margem de erro não superior a 10% em relação a média estimada**.

Em nossa experiência com estudos deste tipo, sabemos que o **custo médio por indivíduo entrevistado fica em torno de R$\$$ 100,00**. Com este conjunto de fatos avalie e obtenha o seguinte conjunto de informações para passar ao cliente:


> 1. Para obter uma estimativa para os parâmetros da população (renda dos chefes de domicílio no Brasil), realize uma amostragem aleatória simples em nosso conjunto de dados. Essa amostra deve conter 200 elementos (utilize random_state = 101 para garantir que o mesmo experimento posso ser realizado novamente). Obtenha a média e o desvio-padrão dessa amostra.
    
> 2. Para a **margem de erro** especificada pelo cliente obtenha os **tamanhos de amostra** necessários para garantir os **níveis de confiança de 90%, 95% e 99%**.
    
> 3. Obtenha o **custo da pesquisa** para os três níveis de confiança.
    
> 4. Para o maior nível de confiança viável (dentro do orçamento disponível), obtenha um **intervalo de confiança para a média da população**.
    
> 5. Assumindo o **nível de confiança escolhido no item anterior**, qual **margem de erro** pode ser considerada utilizando todo o recurso disponibilizado pelo cliente?
    
> 6. Assumindo um **nível de confiança de 95%**, **quanto a pesquisa custaria ao cliente** caso fosse considerada uma **margem de erro de apenas 5%** em relação a média estimada?


# <font color='blue'>Solução do item 1</font>

### Seleção de uma amostra aleatório simples

#### <font color='red'>Lembre-se de utilizar *random_state = 101*</font>

In [ ]:
amostra = dados.sample(n = 200, random_state = 101) # Amostragem com 200 elementos
amostra

UF  Sexo  Idade  Cor  Anos de Estudo  Renda    Altura
29042  29     0     39    8               5    480  1.719128
62672  43     0     55    2               6    250  1.639205
29973  29     1     36    2              12    788  1.654122
22428  26     0     46    8               8   1680  1.622450
55145  41     0     37    2               9   2500  1.625268
...    ..   ...    ...  ...             ...    ...       ...
50497  35     0     39    2              12   3400  1.736021
3668   13     1     66    8               1    780  1.753201
4872   14     1     46    8              12   1000  1.783724
38670  31     0     53    8              12   1000  1.753985
71730  52     1     47    4               9    600  1.667561

[200 rows x 7 columns]

In [ ]:
media = amostra.Renda.mean()
media

1964.205

In [ ]:
desvio_padrao_amostral = amostra.Renda.std()
desvio_padrao_amostral

3139.8855167452157

### Dados do problema

In [ ]:
custo_medio_por_individuo = 100
custo_medio_por_individuo

100

# <font color='blue'>Solução do item 2</font>

### Obtenha a margem de erro

#### <font color='red'>Lembre-se que a margem de erro deve estar na mesma unidade da variável que está sendo estudada (R$)</font>

In [ ]:
e = 0.1 * media # Margem de erro equivale a no máx 10% da média amostral
e

196.4205

### Tamanho da amostra ($1 - \alpha = 90\%$)

In [ ]:
z = norm.ppf(0.5 + (0.90 / 2))
z

1.6448536269514722

In [ ]:
n_90 = int(((z * (desvio_padrao_amostral / e)) ** 2).round())
n_90 # Número de individuos necessários para um nivel de confiança pedido

691

### Tamanho da amostra ($1 - \alpha = 95\%$)

In [ ]:
z = norm.ppf(0.5 + (0.95 / 2))
z

1.959963984540054

In [ ]:
n_95 = int(((z * (desvio_padrao_amostral / e)) ** 2).round())
n_95

982

### Tamanho da amostra ($1 - \alpha = 99\%$)

In [ ]:
z = norm.ppf(0.5 + (0.99 / 2))
z

2.5758293035489004

In [ ]:
n_99 = int(((z * (desvio_padrao_amostral / e)) ** 2).round()) 
n_99 #

1695

# <font color='blue'>Solução do item 3</font>

### Custo da pesquisa para o nível de confiança de 90%

In [ ]:
custo_90 = n_90 * custo_medio_por_individuo # Cálculo do custo : numero de entrevistados * custo por entrevista
custo_90

69100

### Custo da pesquisa para o nível de confiança de 95%

In [ ]:
custo_95 = n_95 * custo_medio_por_individuo
custo_95

98200

### Custo da pesquisa para o nível de confiança de 99%

In [ ]:
custo_99 = n_99 * custo_medio_por_individuo
custo_99

169500

# <font color='blue'>Solução do item 4</font>

In [ ]:
intervalo_95 = norm.interval(alpha = 0.95, loc = media, scale = desvio_padrao_amostral / np.sqrt(n_95))
intervalo_95

(1767.820973280509, 2160.589026719491)

# <font color='blue'>Solução do item 5</font>

In [ ]:
n_95_max = n_95 + ((150000 - custo_95) / 100) 
n_95_max

1500.0

In [ ]:
z_95 = norm.ppf(0.5 + (0.95 / 2))
z_95

1.959963984540054

In [ ]:
e_max = (z_95 * (desvio_padrao_amostral / np.sqrt(n_95_max)))
e_max
print(f'A nova margem de erro utilizando todo o recurso fornecido é de R$ {e_max:.2f}')

A nova margem de erro utilizando todo o recurso fornecido é de R$ 158.90


In [ ]:
margem_erro_percentual = e_max / media * 100
margem_erro_percentual

8.089644982409544

# <font color='blue'>Solução do item 6</font>

In [ ]:
e_5 = 0.05 * media
e_5

98.21025

In [ ]:
z_95 

1.959963984540054

In [ ]:
n_95_novo = int((( z_95 * (desvio_padrao_amostral / e_5)) **2).round())
n_95_novo

3927

In [ ]:
custo_margem_5 = n_95_novo * custo_medio_por_individuo
custo_margem_5

392700

In [ ]:
print(f'O custo para realizar a pesquisa com uma margem de 5% e nível de confiança de 95% é de: R$ {custo_margem_5},00')

O custo para realizar a pesquisa com uma margem de 5% e nível de confiança de 95% é de: R$ 392700,00
